src/backend/distributed/relay/relay_event_utility.c

```diff
@@ -82,17 +82,48 @@ RelayEventExtendNames(Node *parseTree, char *schemaName, uint64 shardId)
 		{
 			AlterObjectSchemaStmt *alterObjectSchemaStmt =
 				(AlterObjectSchemaStmt *) parseTree;
-			char **relationName = &(alterObjectSchemaStmt->relation->relname);
-			char **relationSchemaName = &(alterObjectSchemaStmt->relation->schemaname);
+			ObjectType objectType = alterObjectSchemaStmt->objectType;
 
-			/* prefix with schema name if it is not added already */
-			SetSchemaNameIfNotExist(relationSchemaName, schemaName);
+			if (objectType == OBJECT_STATISTIC_EXT)
+			{
+				RangeVar *stat = makeRangeVarFromNameList(
+					(List *) alterObjectSchemaStmt->object);
+
+				/* append shard id */
+				AppendShardIdToName(&stat->relname, shardId);
```

Comment:
Maybe add a comment:
```c
/*
 * We do not add the schema name to the statistics because it may differ from the
 * relation schema. More importantly, we always construct CREATE STATISTICS with
 * explicit schema name.
 */
```